app/src/main/java/com/google/cloud/solutions/flexenv/common/SpeechTranslationHelper.java

```diff
@@ -0,0 +1,146 @@
+import org.chromium.net.UrlResponseInfo;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+/**
+ * Singleton class that makes requests to a Google Cloud Function that translates speech messages.
+ * For more information about the function implementation, see
+ * https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/master/functions/speech-to-speech
+ */
+public class SpeechTranslationHelper {
+    private static final String TAG = "SpeechTranslationHelper";
+    private static final String SPEECH_TRANSLATE_HTTP_METHOD = "POST";
+    private static final String SPEECH_TRANSLATE_CONTENT_TYPE = "application/json";
+    private static final String SPEECH_TRANSLATE_ENCODING = "LINEAR16";
+
+    private static SpeechTranslationHelper _instance;
+
+    private SpeechTranslationHelper() { }
+
+    public static SpeechTranslationHelper getInstance() {
+        if(_instance == null) {
+            _instance = new SpeechTranslationHelper();
+        }
+        return _instance;
+    }
+
+    /**
+     * Performs a request to a Google Cloud Function that translates speech messages. Returns a JSON
+     * string with information about the response. The response includes information about the audio
+     * files that the client can download at a different time.
+     * @param context The application context
+     * @param base64EncodedAudioMessage The base64-encoded audio message
+     * @param sampleRateInHertz The sample rate in hertz
+     * @param translationListener The callback to deliver the results to.
+     */
+    public void translateAudioMessage(Context context, String base64EncodedAudioMessage,
+                                      int sampleRateInHertz,
+                                      SpeechTranslationListener translationListener) {
+        JSONObject requestBody = new JSONObject();
+        try {
+            requestBody.put("encoding", SPEECH_TRANSLATE_ENCODING);
+            requestBody.put("sampleRateHertz", sampleRateInHertz);
+            requestBody.put("languageCode", context.getResources().getConfiguration().getLocales().get(0));
+            requestBody.put("audioContent", base64EncodedAudioMessage);
+        } catch(JSONException e) {
+            Log.e(TAG, e.getLocalizedMessage());
+            translationListener.onTranslationFailed(e);
+        }
+
+        byte[] requestBodyBytes = requestBody.toString().getBytes();
+        UrlRequest request = buildSpeechTranslationRequest(context, requestBodyBytes, translationListener);
+        request.start();
+    }
+
+    private UrlRequest buildSpeechTranslationRequest(Context context, byte[] requestBody, SpeechTranslationListener translationListener) {
+        CronetEngine.Builder myBuilder = new CronetEngine.Builder(context);
+        CronetEngine cronetEngine = myBuilder.build();
```

Comment:
Oh, I didn't know that. I think I can move this to the activity level. That would solve the same issue in  GCSDownloadHelper too.

Reply:
Implemented in 1ea77bd2d3fbc361df57fa36d623053af2b18537.

In summary:
* Moved the instantiation of `CronetEngine` to the activity, where it can be used by this method and `GcsDownloadHelper.buildGcsRequest()` below.
* The instance is reused on multiple requests until it's discarded by the Android system. For example, the instance has to be recreated on configuration changes such as on screen orientation changes.